### DISCUSSION TITLE
Outbound Email Delivery via Additional Droplets

### DISCUSSION THREAD

**Discussion by tpnsolutions** [postalserver/postal#1632] [Help with using Postal]
I know there is a feature for sending email via "IP Pools" however Digital Ocean doesn't allow for multiple IPs so this option isn't possible. Further, in an effort to design a scalable solution, would it be possible to designate multiple "droplets" each with their own IP address to handle outbound email?

Our current solution implements multiple relay servers which handle delivery of mail in a distributed manner, however one thing I like about Postal is the ability to easily track the status of a message amongst other useful features.

It would be nice to have Postal delegate delivery to an external SMTP server, and still be able to track the status of delivery.

**Comment by willpower232** [comment]
If you have set up the DNS so that SPF trusts all servers involved, you could run only the worker docker container on the additional servers. As long as you are running at least one of each process somewhere and they can all talk to the same RabbitMQ and MySQL/Maria database server then it should be fine.

I'm not sure but you might need to run the SMTP container on each worker server as well so that each server can receive SMTP-related verification checks.

In Digital Ocean speak, you would also want floating IPs on each server to keep the "trust" you earn for each servers IP.

**Reply by tpnsolutions** [reply]
Thanks for the response.

So if I understand correctly, would I be distributing the load across the cluster from the app side or from Postal itself?

Historically when I setup Relays email would send from a central server to one of the relay nodes which ultimately sent to the end user. When this was done because trust was built into the design between main server and relay nodes the main servers logs would always show "status=sent" while if the end user server bounced or deferred delivery the relay nodes was the only one aware of this status.

I want postal to be aware of any failure sent from end user (destination) server ultimately.

**Reply by willpower232** [reply]
You would be manually distributing the load of Postal, each process doesn't really know or care about the others, they just stick to their job.

As everything is a part of Postal, any information would be centralised. As long as your database and queue are fault tolerant and reliable, it should be pretty good.

Because of the existence of docker and containers, you're essentially creating a manual kubernetes cluster 😅

**Reply by tpnsolutions** [reply]
@willpower232,

Getting back to the topic, I have a few more clarifying questions.

So, from what we've established.

1. Setup global database server

2. Setup global rabbitmq server

3. Download postal on all droplet

4. Run "postal bootstrap hostname" on all droplets

5. Configure postal to use "global database" and "global rabbitmq" server on all droplets

6. Run "postal initialize" and "postal make-user" on one droplet

7. Run "postal start" on all droplets"

8. Setup web proxy on one or all droplets.

Did I get this right???

Thanks Will, and happy holidays!

**Reply by willpower232** [reply]
Sounds good to me, in this scenario all your droplets would be sending emails so either you would need to do something specific to route your outgoing messages via a specific IP or list all the IP addresses in some of the DNS records to make sure deliverability was consistently good across your whole deployment.

**Reply by tpnsolutions** [reply]
Yeah, I'd likely setup as I have done in other deployments (without postal) a common subdomain like: relay.domain.com with each postal IP address listed using that subdomain. Then setup the originating server to route mail through relay.domain.com which would do a sort of "round-robin" style routing.

For destinations that present a problem, I'd likely establish for instance "hotmail-com.relay.domain.com" and dynamically adjust the DNS zone to only include postal IPs that are NOT blocked while the issue gets addressed.

The benefit I see with postal, amongst others is that I can create a Webhook to let me know when a problem occurs, and even trigger the custom zone updates (ex. hotmail-com.relay.domain.com).
